src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -4848,6 +4853,21 @@ internal void CleanupWindowsPowerShellCompatResources()
             }
         }
 
+        internal static void SyncCurrentLocationHandler(object sender, LocationChangedEventArgs args)
+        {
+            PSSession compatSession = GetWindowsPowerShellCompatRemotingSession();
+            if (compatSession?.Runspace.RunspaceStateInfo.State == RunspaceState.Opened)
```

Comment:
@PaulHigin Is this the right way to check if remote connection is broken ?

Reply:
Yes, this is correct.  If the session is not in the 'Opened' state then it cannot run commands.